nearcache/nearcache_it_test.go

```diff
@@ -325,5 +448,42 @@ func newNearCacheMapTestContext(t *testing.T, fmt nearcache.InMemoryFormat, inva
 			tcx.Config.AddNearCache(ncc)
 		},
 	}
+}
 
+func assertNearCacheExpiration(tcx it.MapTestContext, size int32) {
+	t := tcx.T
+	m := tcx.M
+	it.Eventually(t, func() bool {
+		nca := hz.MakeNearCacheAdapterFromMap(m).(it.NearCacheAdapter)
+		stats := m.LocalMapStats().NearCacheStats
+		// make assertions over near cache's backing map size.
+		t.Logf(
+			"size: %d, OEC: %d, OEMC: %d, ex: %d, ev: %d",
+			nca.Size(),
+			stats.OwnedEntryCount,
+			stats.OwnedEntryMemoryCost,
+			stats.Expirations,
+			stats.Evictions,
+		)
+		if nca.Size() != 0 {
+			return false
+		}
+		// make assertions over near cache stats.
+		if stats.OwnedEntryCount != 0 {
+			return false
+		}
+		if !assert.Equal(t, int64(0), stats.OwnedEntryMemoryCost) {
+			return false
+		}
+		if stats.Expirations != int64(size) {
+			return false
+		}
+		if stats.Evictions != 0 {
+			return false
+		}
+		if !assert.Equal(t, int64(0), stats.Evictions) {
+			return false
+		}
```

Comment:
These two if looks duplicates to me.

Reply:
https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/d6d75b4566197995c50ed6a08eab6176e2479712